XenAdmin/Dialogs/PvsFarmDialog.cs

```diff
@@ -0,0 +1,127 @@
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR 
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING 
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF 
+ * SUCH DAMAGE.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.ComponentModel;
+using XenAdmin.Network;
+using XenAdmin.Controls.DataGridViewEx;
+using XenAPI;
+
+namespace XenAdmin.Dialogs
+{
+    public partial class PvsFarmDialog : XenDialogBase
+    {
+        /// <summary>
+        /// Creates a dialog for viewing the PVS farms and PVS servers on a particular connection.
+        /// </summary>
+        /// <param name="_connection">May not be null.</param>
+        public PvsFarmDialog(IXenConnection _connection)
+        {
+            System.Diagnostics.Trace.Assert(_connection != null);
+            connection = _connection;
+
+            InitializeComponent();
+            Text += string.Format(" - '{0}'", this.connection.Name);
+
+            System.Diagnostics.Trace.Assert(gridView.Columns.Count > 0);
+            gridView.Columns[0].DefaultCellStyle.NullValue = null;
+
+            RegisterEventHandlers();
+            Rebuild();
+        }
+
+        private void RegisterEventHandlers()
+        {
+            UnregisterEventHandlers();
+            connection.Cache.RegisterBatchCollectionChanged<PVS_farm>(PvsFarmBatchCollectionChanged);
+            connection.Cache.RegisterBatchCollectionChanged<PVS_server>(PvsServerBatchCollectionChanged);
+        }
+
+        private void UnregisterEventHandlers()
+        {
+            connection.Cache.DeregisterBatchCollectionChanged<PVS_farm>(PvsFarmBatchCollectionChanged);
+            connection.Cache.DeregisterBatchCollectionChanged<PVS_server>(PvsServerBatchCollectionChanged);
+        }
+
+        private void PvsFarmBatchCollectionChanged(object sender, EventArgs e)
+        {
+            Program.Invoke(this, Rebuild);
+        }
+
+        private void PvsServerBatchCollectionChanged(object sender, EventArgs e)
+        {
+            Program.Invoke(this, Rebuild);
+        }
+
+        protected override void OnClosing(CancelEventArgs e)
+        {
+            UnregisterEventHandlers();
+
+            base.OnClosing(e);
+        }
+
+        private void Rebuild()
+        {
+            Program.AssertOnEventThread();
+
+            try
+            {
+                gridView.SuspendLayout();
+                gridView.Rows.Clear();
+
+                PVS_farm[] pvsFarmsArray = connection.Cache.PVS_farms;

```

Comment:
We can use Sort() function here, which will sort the XenObjets by Name property (which we will need to define). E.g.
`var pvsFarms = new List<PVS_farm>(Connection.Cache.PVS_farms);`
`pvsFarms .Sort();`